lib/view_component/test_helpers.rb

```diff
@@ -162,5 +173,71 @@ def with_request_url(path)
     def build_controller(klass)
       klass.new.tap { |c| c.request = request }.extend(Rails.application.routes.url_helpers)
     end
+
+    # This is a simpler version of {Capybara::Session}.
+    #
+    # It only includes {Capybara::Node::Finders}, {Capybara::Node::Matchers},
+    # {#within} and {#within_element}. It is useful in that it does not require a
+    # session, an application or a driver, but can still use Capybara's finders
+    # and matchers on any string that contains HTML.
+    #
+    # Most of this code is shamelessly stolen from the {Capybara::Session} class
+    # in the Capybara gem (https://github.com/teamcapybara/capybara/blob/e704d00879fb1d1e1a0cc01e04c101bcd8af4a68/lib/capybara/session.rb#L38).
+    class SimpleSession
+      # Stolen from: https://github.com/teamcapybara/capybara/blob/e704d00879fb1d1e1a0cc01e04c101bcd8af4a68/lib/capybara/session.rb#L767-L774.
+
+      Capybara::Session::NODE_METHODS.each do |method|
+        if RUBY_VERSION >= '2.7'
+          class_eval <<~METHOD, __FILE__, __LINE__ + 1
+            def #{method}(...)
+              @touched = true
+              current_scope.#{method}(...)
+            end
+          METHOD
+        else
+          define_method method do |*args, &block|
+            @touched = true
+            current_scope.send(method, *args, &block)
+          end
+        end
+      end
+
+      # Initializes the receiver with the given string of HTML.
+      #
+      # @param html [String] the HTML to create the session out of
+      def initialize(html)
+        @document = Capybara::Node::Simple.new(html)
+      end
+
+      def within(*args, **kw_args)
+        new_scope = args.first.respond_to?(:to_capybara_node) ? args.first.to_capybara_node : find(*args, **kw_args)
+        begin
+          scopes.push(new_scope)
+          yield if block_given?
+        ensure
+          scopes.pop
+        end
+      end
+
+      alias_method :within_element, :within
+
+      def native
```

Comment:
This method was needed by the PVC tests. But this method does not exist on the Capybara session of a regular browser based test.